pay/contracts/subscription/HectorSubscription.sol

```diff
@@ -0,0 +1,498 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+pragma solidity ^0.8.17;
+
+import {IERC20} from '@openzeppelin/contracts/token/ERC20/IERC20.sol';
+import {SafeERC20} from '@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol';
+import {OwnableUpgradeable} from '@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol';
+import {ReentrancyGuardUpgradeable} from '@openzeppelin/contracts-upgradeable/security/ReentrancyGuardUpgradeable.sol';
+
+interface Factory {
+    function parameter() external view returns (bytes memory);
+
+    function owner() external view returns (address);
+}
+
+error INVALID_PARAM();
+error INVALID_ADDRESS();
+error INVALID_AMOUNT();
+error INVALID_TIME();
+error INVALID_PLAN();
+error PAYER_IN_DEBT();
+error INACTIVE_SUBSCRIPTION();
+error ACTIVE_SUBSCRIPTION();
+error INVALID_MODERATOR();
+
+contract HectorSubscription is OwnableUpgradeable, ReentrancyGuardUpgradeable {
+    using SafeERC20 for IERC20;
+
+    /* ======== STORAGE ======== */
+
+    struct Plan {
+        address token; // TOR, WFTM
+        uint48 period; // 3 months, 6 months, 12 months
+        uint256 amount;
+    }
+
+    struct Subscription {
+        uint256 planId;
+        uint48 expiredAt;
+    }
+
+    /// @notice product
+    string public product;
+
+    /// @notice treasury wallet
+    address public treasury;
+
+    /// @notice subscription plans configurable by admin
+    Plan[] public plans;
+
+    /// @notice expire deadline to cancel subscription
+    uint48 public expireDeadline = 30 days;
```

Comment:
@bejaxer please make this expireDeadline configurable

Reply:
by moderator or by admin?

Reply:
both is fine.